packages/block-library/src/cover/test/edit.js

```diff
@@ -416,13 +416,13 @@ describe( 'Cover block', () => {
 				} )
 			);
 			await userEvent.click( screen.getByText( 'Overlay' ) );
-			// The default color is black, so clicking the black color option will remove the background color,
-			// which should remove the isDark setting and assign the is-light class.
+			// The default color is black, so clicking the black color option will remove the background color.
+			// The fallback CSS is still black, so the is-dark class should remain.
 			const popupColorPicker = screen.getByRole( 'button', {
 				name: 'Color: Black',
 			} );
 			await userEvent.click( popupColorPicker );
-			expect( coverBlock ).toHaveClass( 'is-light' );
+			expect( coverBlock ).not.toHaveClass( 'is-light' );
```

Comment:
@glendaviesnz I think the behavior tested here was wrong. The default value in cover block CSS is black, so the block should have the `is-dark-theme` class, not `is-light` if there is no color manually set.

Reply:
Glen must agree because it was fixed in https://github.com/WordPress/gutenberg/pull/53253/commits/56f6bc53e7057968a550391c7e3f6e0bab12e3f8#diff-a98ee8d5e36c3c5d47d887be50f9058b8dff86e6b475d000b4a2c7b80c1e6888L407-R435.

I'm not sure this version is fixed though. Nothing here causes changes to the classes between the assertions and the assertions are the same. It looks like the test didn't fail before 7f87feb was reverted and it should have by intent.

Reply:
I've gone ahead and created #53375 from this branch. It includes 0b6eb9a, changing this test to work as was done by Glen. It also includes 08ebc9d fixes for a couple other failing tests (the same two that are still failing on this branch at present due to `act` warnings).

If you'd like I'd be glad to push just those changes to this branch. In case you agree and don't want to wait for me here’s all it should take:
```
git fetch upstream fix/cover-is-dark-flicker1a
git cherry-pick 08ebc9d 0b6eb9a
```

Reply:
> @glendaviesnz I think the behavior tested here was wrong. 

Yes, it was wrong. Those tests were testing the existing behaviour in trunk, which was in fact broken, and as @stokesman notes, the broken behaviour was fixed in #53253 and the test updated.